### ISSUE TITLE
Support for SPSS multiple response categories

### ISSUE THREAD

**Issue by Berndvanderwielen** [Roche/pyreadstat#25]
Support to retrieve (meta) data on MRVs / multiple answer question groupings would be great.

**Comment by ofajardo** [comment]
Sorry, no idea what MRVs are. Can you please provide an example SPSS file and explain what it is and what information are you trying to retrieve?

**Comment by Berndvanderwielen** [comment]
The official name is "Multiple Response sets". URL: https://www.ibm.com/support/knowledgecenter/en/SSLVMB_23.0.0/spss/base/multiple_response_intro.html

If the SPSS documentation is not enough I can provide an example SPSS file.

**Comment by ofajardo** [comment]
Yes, a sample file will be needed. In addition a description of the contents in plain text, because the job is to guess where in the binary file is the content you are looking for.

This will require changes to the Readstat C library. I can file an issue over there or you can do it yourself if you prefer. There is no guarantee that they will do it, nor timelines either. 

It will help a lot if there is somewhere a description of how are these fields represented in the binary file. If you could find that would be great, because otherwise it will be very difficult to implement.

**Comment by ofajardo** [comment]
Closed due to lack of example files.

**Comment by SamMousa** [comment]
@ofajardo, sorry for the ping, but I assume you're not receiving comments on closed issues.

Could this be reopened?

The specs for this record in the SPSS file are actually part of the spec you linked: https://www.gnu.org/software/pspp/pspp-dev/html_node/Multiple-Response-Sets-Records.html

Essentially what it does is specify the relationship between multiple questions that should be interpreted as a single question with multiple values instead.
[example.sav.zip](https://github.com/Roche/pyreadstat/files/5511176/example.sav.zip)

Attaches is an example file that contains 2 sets, one multiple category and multiple dichotomy. For details you could check docs here (https://www.gnu.org/software/pspp/manual/html_node/MRSETS.html#MRSETS), but for implementation that is not relevant.


Let me know if I can be of further assistance, I'm not familiar with python at all, but have spent many hours hating the binary file format that is SPSS SAV...

**Comment by ofajardo** [comment]
hi there, 

Haven't look into it in detail yet, but this will require that [Readstat](https://github.com/WizardMac/ReadStat) (the C library behind pyreadstat) implements this. 

Could you therefore open an issue there? (I am sure they will appreciate your insights into the binary file format). Once it is implemented in Readstat I will be able to bring it into pyreadstat.

**Comment by slobodan-ilic** [comment]
I've opened a new PR to address this #259 . In accordance with our team at Crunch.io and Evan Miller. We'll also open a PR on readstat, so this won't be immediately available. The idea is to rebase the ☝️ pr once readstat changes get shipped.

**Comment by arsoni20** [comment]
This feature will be hugely appreciated.

**Comment by arsoni20** [comment]
Thanks a lot 🥇
